core/src/main/scala/cats/syntax/list.scala

```diff
@@ -31,24 +32,95 @@ final class ListOps[A](private val la: List[A]) extends AnyVal {
+   * Groups elements inside this `List` according to the `Order` of the keys
+   * produced by the given key function.
+   * And each element in a group is transformed into a value of type B
+   * using the mapping function.
    *
+   * {{{
+   * scala> import scala.collection.immutable.SortedMap
+   * scala> import cats.data.NonEmptyList
+   * scala> import cats.implicits._
    * scala> val list = List(12, -2, 3, -5)
+   * scala> val expectedResult = SortedMap(false -> NonEmptyList.of("-2", "-5"), true -> NonEmptyList.of("12", "3"))
+   * scala> list.groupMapNel(_ >= 0)(_.toString) === expectedResult
+   * res0: Boolean = true
+   * }}}
+   */
+  def groupMapNel[K, B](key: A => K)(f: A => B)(implicit K: Order[K]): SortedMap[K, NonEmptyList[B]] = {
+    implicit val ordering: Ordering[K] = K.toOrdering
+    toNel.fold(SortedMap.empty[K, NonEmptyList[B]])(_.groupMap(key)(f))
+  }
+
+  /**
+   * Groups elements inside this `List` according to the `Order` of the keys
+   * produced by the given key function.
+   * Then each element in a group is transformed into a value of type B
+   * using the mapping function.
+   * And finally they are all reduced into a single value
+   * using their `Semigroup`.
    *
-   * scala> val expectedResult = SortedMap(false -> NonEmptyList.of(-2, -5), true -> NonEmptyList.of(12, 3))
+   * {{{
+   * scala> import scala.collection.immutable.SortedMap
+   * scala> import cats.implicits._
+   * scala> val list = List("Hello", "World", "Goodbye", "World")
+   * scala> val expectedResult = SortedMap("goodbye" -> 1, "hello" -> 1, "world" -> 2)
+   * scala> list.groupMapReduce(_.strip.toLowerCase)(_ => 1) === expectedResult
+   * res0: Boolean = true
+   * }}}
+   */
+  def groupMapReduce[K, B](key: A => K)(f: A => B)(implicit K: Order[K], B: Semigroup[B]): SortedMap[K, B] =
+    groupMapReduceWith(key)(f)(B.combine)
+
+  /**
+   * Groups elements inside this `List` according to the `Order` of the keys
+   * produced by the given key function.
+   * Then each element in a group is transformed into a value of type B
+   * using the mapping function.
+   * And finally they are all reduced into a single value
+   * using the provided combine function.
    *
-   * scala> list.groupByNel(_ >= 0) === expectedResult
+   * {{{
+   * scala> import scala.collection.immutable.SortedMap
+   * scala> import cats.implicits._
+   * scala> val list = List("Hello", "World", "Goodbye", "World")
+   * scala> val expectedResult = SortedMap("goodbye" -> 1, "hello" -> 1, "world" -> 2)
+   * scala> list.groupMapReduceWith(_.strip.toLowerCase)(_ => 1)(_ + _) === expectedResult
    * res0: Boolean = true
    * }}}
    */
-  def groupByNel[B](f: A => B)(implicit B: Order[B]): SortedMap[B, NonEmptyList[A]] = {
-    implicit val ordering: Ordering[B] = B.toOrdering
-    toNel.fold(SortedMap.empty[B, NonEmptyList[A]])(_.groupBy(f))
+  def groupMapReduceWith[K, B](key: A => K)(f: A => B)(combine: (B, B) => B)(implicit K: Order[K]): SortedMap[K, B] = {
```

Comment:
Not sure if it is worth it to add this method.

It is mostly the same as the one in the stdlib, just that this returns a **SortedMap**.
_(And that it is available in `2.12`)_